repos/system_upgrade/el7toel8/actors/removeupgradebootentry/libraries/library.py

```diff
@@ -1,16 +1,27 @@
 from leapp.exceptions import StopActorExecutionError
 from leapp.libraries.stdlib import CalledProcessError, api, run
-from leapp.models import BootContent
+from leapp.models import BootContent, SystemFacts
 
 
 def remove_boot_entry():
     # we need to make sure /boot is mounted before trying to remove the boot entry
+
+    facts_msg = api.consume(SystemFacts)
+    facts = next(facts_msg, None)
+    if not facts:
+        raise StopActorExecutionError('Could not identify system firmware',
+                                      details={'details': 'Actor did not receive SystemFacts message.'})
+
+    mount_points_per_firmware = {
+        'bios': ['/boot'],
+        'efi': ['/boot', '/boot/efi']
+    }
+
     try:
-        run([
-            '/bin/mount', '/boot'
-        ])
+        for mp in mount_points_per_firmware[facts.firmware]:
+            run(['/bin/mount'] + mp)
     except CalledProcessError:
-        # /boot has been most likely already mounted
+        # partitions have been most likely already mounted
```

Comment:
I don't think we should do it like this, but have the exception being caught in the loop, just in case one of the given mount points is not a mount point.